src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -37,13 +37,15 @@ void CoreWorkerDirectTaskSubmitter::AddWorkerLeaseClient(
 }
 
 void CoreWorkerDirectTaskSubmitter::OnWorkerIdle(
-    const rpc::WorkerAddress &addr, const SchedulingKey &scheduling_key, bool was_error,
+    const rpc::WorkerAddress &addr, const SchedulingKey &scheduling_key,
+    bool is_actor_creation, bool was_error,
     const google::protobuf::RepeatedPtrField<rpc::ResourceMapEntry> &assigned_resources) {
   auto lease_entry = worker_to_lease_client_[addr];
   auto queue_entry = task_queues_.find(scheduling_key);
   // Return the worker if there was an error executing the previous task,
+  // the previous task is an actor creation task,
   // there are no more applicable queued tasks, or the lease is expired.
-  if (was_error || queue_entry == task_queues_.end() ||
+  if (was_error || is_actor_creation || queue_entry == task_queues_.end() ||
```

Comment:
For actor creation we don't want to return the worker, since it is now used by the actor indefinitely.

Reply:
I see your point, I was intentionally returning to raylet so that we can reuse the logic in raylet to handle actor creation